### PR TITLE
Use a generic message for api auth failure

### PR DESCRIPTION
**Pull request by alanhartless**
[//]: # ( Please answer the following questions: )

| Q  | A
| --- | ---
| Bug fix? | Y
| New feature? | 
| Related user documentation PR URL | 
| Related developer documentation PR URL | 
| Issues addressed (#s or URLs) | 
| BC breaks? | 
| Deprecations? | Y

[//]: # ( Note that all new features should have a related user and/or developer documentation PR in their respective repositories. )

[//]: # ( Required: )
#### Description:

Because oauth2 is the last firewall to be checked for API access, it's response of `{"error":"access_denied","error_description":"OAuth2 authentication required"}` is sent even though oauth2 may not be used which leads to confusion since basic auth or oauth1 could be attempted. This is now replaced with a generic response of `{"errors":[{"message":"API authorization denied.","code":401,"type":"access_denied"}],"error":"access_denied","error_description":"API authorization denied. (`error` and `error_description` are deprecated as of 2.6.0 and will be removed in 3.0. Use the `errors` array instead.)"}`.

#### Steps to test this PR:
1. Make a call to an API url without authentication and you should see the new response.

[//]: # ( As applicable: )
#### Steps to reproduce the bug:
1. Same as above only you'll see the oauth2 message.

#### List deprecations along with the new alternative:
1. `error` and `error_description` are deprecated as they were only used in the context of OAuth2. The new `errors` array will replace it to allow consistency across all API errors.

### CODE REVIEW

**Comment by MarkLL** [comment]
I like the idea in principal but it could break existing API's. Maybe it's time to add versioning to the API as well so we can track the expected response and not break anyone else's implementation. Just my 2c's

**Comment by dongilbert** [comment]
@MarkLL It shouldn't break any existing API implementations, as the returned JSON still includes the previous structure, which can be accessed in the same way it was before.

**Comment by MarkLL** [comment]
@dongilbert until version 3 when it removed! the main reason for my comment is to push for an API Library version so the Mautic server can be a bit more forgiving. see current conversation on slack...

File: app/bundles/ApiBundle/Controller/CommonApiController.php [hunk 1]
```diff
@@ -674,11 +674,20 @@ protected function returnError($msg, $code, $details = [])
...
+                    [
+                        'message' => $this->get('translator')->trans($msg, [], 'flashes'),
+                        'code'    => $code,
+                        'type'    => null,
+                        'details' => $details,
+                    ],
+                ],
+                // @deprecated 2.6.0 to be removed in 3.0
                 'error' => [
+                    'message' => $this->get('translator')->trans($msg, [], 'flashes').' (`error` is deprecated as of 2.6.0 and will be removed in 3.0. Use the `errors` array instead.)',,
```
Comment (dongilbert) [inline app/bundles/ApiBundle/Controller/CommonApiController.php]: Pushed an update to fix this error I was receiving due to double comma at eol

```
Parse error: syntax error, unexpected ',', expecting ']' in /Users/dgilbert/Sites/mautic/app/bundles/ApiBundle/Controller/CommonApiController.php on line 687
```

**Pull request merged**